### PR TITLE
fix: proxy contract selection

### PR DESCRIPTION
**Pull request by MatejVukosav**

### CODE REVIEW

File: crates/context/src/lib.rs [hunk 1]
```diff
@@ -1078,14 +1078,19 @@ impl ContextManager {
...
         };
 
+        let proxy_contract = self.get_proxy_contract(context_id).await?;
         let _ = self
             .config_client
             .mutate::<ContextProxy>(
                 context_config.protocol.as_ref().into(),
                 context_config.network.as_ref().into(),
-                context_config.contract.as_ref().into(),
+                proxy_contract.into(),
```
Comment (frdomovic) [inline crates/context/src/lib.rs]:
```suggestion
                context_config.proxy_contract.as_ref().into(), 
```

File: crates/context/src/lib.rs [hunk 2]
```diff
@@ -1078,14 +1078,19 @@ impl ContextManager {
             bail!("No private key found for signer");
         };
 
+        let proxy_contract = self.get_proxy_contract(context_id).await?;
```
Comment (frdomovic) [inline crates/context/src/lib.rs]:
```suggestion
```

File: crates/context/src/lib.rs (hunk 1 above)
Comment (frdomovic) [reply crates/context/src/lib.rs]: We already have it saved in config we should not refetch this @miraclx

File: crates/context/src/lib.rs [hunk 3]
```diff
@@ -1111,14 +1116,15 @@ impl ContextManager {
...
```
Comment (frdomovic) [inline crates/context/src/lib.rs]:
```suggestion
```

File: crates/context/src/lib.rs [hunk 4]
```diff
@@ -1111,14 +1116,15 @@ impl ContextManager {
...
         let _ = self
             .config_client
             .mutate::<ContextProxy>(
                 context_config.protocol.as_ref().into(),
                 context_config.network.as_ref().into(),
-                context_config.contract.as_ref().into(),
+                proxy_contract.into(),
```
Comment (frdomovic) [inline crates/context/src/lib.rs]:
```suggestion
                 context_config.proxy_contract.as_ref().into(), 
```

**Review by frdomovic — APPROVED** [review]

**Review by miraclx — APPROVED** [review]

**Pull request merged**